 --- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch84/Time_11/DateTimeZoneBuilder/Time_11_DateTimeZoneBuilder_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch84/Time_11/DateTimeZoneBuilder/Time_11_DateTimeZoneBuilder_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -369,6 +369,7 @@
                 millis = next.getMillis();
                 saveMillis = next.getSaveMillis();
                 if (tailZone == null && i == ruleSetCount - 1) {
+                    if(!((ruleSetCount <= 1) && ((tailZone!=null) || (outputID))))
                     tailZone = rs.buildTailZone(id);
                     // If tailZone is not null, don't break out of main loop until
                     // at least one more transition is calculated. This ensures a
